orttraining/orttraining/python/training/orttrainer.py

```diff
@@ -1202,8 +1203,91 @@ def load_state_dict(self, state_dict, strict=True):
+        """Checks if aggregation is required for the loading the state_dict into the ORTTrainer"""
+
+        # There are three conditions that require aggregation for ZeRO:
+        # - Going from zero distributed run to a non distributed run
+        # - Going from one world size to another (weights need to be aggregated and then again sharded)
+        # - Going from mixed precision to full precision run (since fp32 states are sharded in mixed
+        #   precision and not in full precision)
+        if loaded_trainer_options['zero_stage'] > 0:
+            if self.options.distributed.deepspeed_zero_optimization.stage != loaded_trainer_options['zero_stage'] or \
+                self.options.distributed.world_size != loaded_trainer_options['world_size'] or \
+                    (loaded_trainer_options['mixed_precision'] and not self.options.mixed_precision.enabled):
+                return True
+        return False
+
+    def load_checkpoint(self, *paths, strict=True):
+        """Loads the saved checkpoint state dictionary into the ORTTrainer
+
+        Reads the saved checkpoint files specified by paths from disk and loads the state dictionary
+        onto the ORTTrainer.
+        Aggregates the checkpoint files if aggregation is required.
+
+        Args:
+            paths: one or more files represented as strings where the checkpoint is saved
+            strict: boolean flag to strictly enforce that the saved checkpoint state_dict
+                keys match the keys from ORTTrainer.state_dict
+        Returns:
+            dictionary that the user had saved when calling save_checkpoint
+        """
+        state_dict = {}
+
+        # check if aggregation is required
+        loaded_trainer_options = _checkpoint_storage.load(paths[0], key=_utils.state_dict_trainer_options_key())
+        if self._aggregation_required(loaded_trainer_options):
+            # if aggregation is required, aggregation logic must be run on the saved checkpoints
+            state_dict = checkpoint.aggregate_checkpoints(paths, pytorch_format=False)
+        else:
+            # if aggregation is not required, reorder the checkpoints in order of ascending rank,
+            # and load the checkpoint for the current ORTTrainer rank.
+            ordered_paths = checkpoint._order_paths(paths)
+            state_dict = _checkpoint_storage.load(ordered_paths[self.options.distributed.world_rank])
+            assert state_dict['trainer_options']['world_rank'] == (self.options.distributed.world_rank), \
+                "Loaded checkpoint rank does not match the trainer rank."
+
+        # extract user dict from the saved checkpoint
+        user_dict = {}
+        if _utils.state_dict_user_dict_key() in state_dict:
+            user_dict = _checkpoint_storage.from_serialized_hex(state_dict[_utils.state_dict_user_dict_key()])
+            del state_dict[_utils.state_dict_user_dict_key()]
+
+        self.load_state_dict(state_dict, strict=strict)
```

Comment:
before this call, we must check if the checkpoint being loaded has a matching optimizer to the current training session using the trainer info.

Reply:
I think this check might be better inside ```load_state_dict``` function. Will add it there.

Reply:
Why cant I see the original comment? I just see yur responses :-S

Reply:
Not sure. I am able to see all the comments on the thread.